### PR TITLE
fetch proxy from bootstrap

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/holochain/pull/1236

### Summary

- actually makes use of the network proxy config to fetch proxy server from the bootstrap service.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by freesig — APPROVED** [review]
Looks good

**Review by thedavidmeister — APPROVED** [review]

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 1]
```diff
@@ -47,6 +47,7 @@ maplit = { version = "1", optional = true }
 mockall = { version = "0.10.2", optional = true }
 
 [dev-dependencies]
+kitsune_p2p_bootstrap = { version = "0.0.3", path = "../bootstrap" }
```
Comment (steveeJ) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:50]: do we need the version here? there's a note in the [cargo book on development dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies):

> Note: When a package is published, only dev-dependencies that specify a version will be included in the published crate. For most use cases, dev-dependencies are not needed when published, though some users (like OS packagers) may want to run tests within a crate, so providing a version if possible can still be beneficial.

**Review by steveeJ — COMMENTED** [review]
i only came here to check out the dev-dependency declaration :wink:

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml (hunk 1 above)
Comment (steveeJ) [reply crates/kitsune_p2p/kitsune_p2p/Cargo.toml:50]:
```suggestion
kitsune_p2p_bootstrap = { path = "../bootstrap" }
```

Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/Cargo.toml:50]: Good point, I don't think we need the version.

File: crates/holochain/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,18 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+- Init functions can now make zome calls. [#1186](https://github.com/holochain/holochain/pull/1186)
```
Comment (steveeJ) [inline crates/holochain/CHANGELOG.md:6]: as you noted on #1258 this addition doesn't belong here. it's duplicated from the most recent release. is this a case of pressing the "merge base into this branch" duplicating changelog items?

**Review by steveeJ — CHANGES_REQUESTED** [review]

File: crates/holochain/CHANGELOG.md (hunk 2 above)
Comment (neonphog) [reply crates/holochain/CHANGELOG.md:6]:
```suggestion
```

I did a merge with the git cli tool, not sure how this happened... it must have made it look like that line should be there, regardless, I'll delete it.

**Pull request merged**